jena-jdbc/jena-jdbc-core/pom.xml

```diff
@@ -39,17 +39,28 @@
 			<version>3.10.0-SNAPSHOT</version>
 		</dependency>
 
-    <!-- Jena Driver configures Log4j directly -->
-    <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-log4j12</artifactId>
-    </dependency>
+		<!-- Jena Driver configures Log4j directly -->
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-log4j12</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>log4j</groupId>
+			<artifactId>log4j</artifactId>
+		</dependency>
+
+		<dependency>
```

Comment:
Why are you pulling in the `log4j` API?

Reply:
This being said, what would you rather see? This issue is titled _Jena master does not build under JDK10_ so I wonder if getting Jena to build under JDK 10 is the immediate response. Someone can then go and work on the underlying logging issue. Right now I have not investigated it enough or else I would have provided the fix.

Reply:
`JenaDriver` can optionally configure Log4j, this is because JDBC drivers can be used in contexts where the end user can't directly configure logging e.g. when dropped into other applications.  In those contexts users can add `logging=<log4j-config>` to have the driver configure logging for them.

However the default behaviour is to not configure anything and all the code internally uses SLF4J so in environments where users can control logging configuration they are free to do that.  So in environments where the user fully controls things they can configure logging as appropriate.

Reply:
Ok thanks @rvesse. The logic seems clear from what you wrote, it doesn't however take away from the fact that the codebase will not build under JDK10 (and possibly JDK9) without the presence of the log4j library containing org.apache.log4j.PropertyConfigurator as currently used within jena/jena-jdbc/jena-jdbc-core/src/main/java/org/apache/jena/jdbc/JenaDriver.java.
The issue here is to either refactor the logging or to add the dependency... I would think there is no other scenario under consideration.
Is that a fair statement?